pkg/roachpb/metadata_replicas.go

```diff
@@ -353,18 +370,36 @@ func (d ReplicaDescriptors) CanMakeProgress(liveFunc func(descriptor ReplicaDesc
 		}
 	}
 
-	votersOldGroup := d.Filter(isVoterOldConfig)
-	liveVotersOldGroup := d.Filter(isBoth(isVoterOldConfig, liveFunc))
+	// This functions handles regular, or joint-consensus replica groups. In the
+	// joint-consensus case, we'll independently consider the health of the
+	// outgoing group ("old") and the incoming group ("new"). In the regular case,
+	// the two groups will be identical.
+
+	votersOldGroup := d.Filter(ReplicaDescriptor.IsVoterOldConfig)
+	liveVotersOldGroup := d.Filter(isBoth(ReplicaDescriptor.IsVoterOldConfig, liveFunc))
 
 	n := len(votersOldGroup)
 	// Empty groups succeed by default, to match the Raft implementation.
-	if n > 0 && len(liveVotersOldGroup) < n/2+1 {
-		return false
-	}
+	availableOutgoingGroup := (n == 0) || (len(liveVotersOldGroup) >= n/2+1)
 
-	votersNewGroup := d.Filter(isVoterNewConfig)
-	liveVotersNewGroup := d.Filter(isBoth(isVoterNewConfig, liveFunc))
+	votersNewGroup := d.Filter(ReplicaDescriptor.IsVoterNewConfig)
+	liveVotersNewGroup := d.Filter(isBoth(ReplicaDescriptor.IsVoterNewConfig, liveFunc))
 
 	n = len(votersNewGroup)
-	return len(liveVotersNewGroup) >= n/2+1
+	availableIncomingGroup := len(liveVotersNewGroup) >= n/2+1
+
+	res.Available = availableIncomingGroup && availableOutgoingGroup
+
+	if replicationFactor == -1 {
```

Comment:
Can you, instead of this special case, use `1` at the one caller and ignore the undesired fields? This effectively removes all special casing which is good.